### DISCUSSION TITLE
How to make ANSI dim/faint text render as lower opacity?

### DISCUSSION THREAD

**Discussion by justmoon** [wezterm/wezterm#4026] [Q&A] [answered]
First off, Wezterm is incredible! It has sensible defaults so I'm able to get a very nice terminal emulator experience with a [pretty minimal configuration](https://github.com/justmoon/dotfiles/blob/main/private_dot_config/wezterm/wezterm.lua).

There is one pet peeve which I haven't been able to solve. In other terminal emulators like VS Code's terminal, the ANSI dim/faint text (SGR 2) is indicated with a lower opacity, whereas in Wezterm it's done via a lighter weight font. For whatever reason, with my combination of font choice, Wezterm's rendering etc., it just doesn't look very nice. It also means that Wezterm can't render dim-and-bold text, so you lose some context sometimes.

Here is an example from VS Code:

![image](https://github.com/wez/wezterm/assets/53233/e85b718f-16ea-4403-be2d-7cd5df086359)

And here is the same thing in Wezterm:

![image](https://github.com/wez/wezterm/assets/53233/3ebd55e5-40d4-4415-811e-4aa311eb9de4)

Note the file sizes stand out in the VS Code rendering but don't in the Wezterm version. I understand that SGR 1;2 is self-contradictory since it's asking the terminal emulator to simultaneously increase and decrease the intensity of this text, but from a practical standpoint: a) many terminal emulators support it, b) many CLI tools use it, and c) it does give some extra flexibility for CLI designers.

Based on searching other discussions on the subject, it seems that you should be able to use font_rules to cause Wezterm to use opacity to indicate dim/faint-ness. However, after several hours of reading the docs ([ex](https://wezfurlong.org/wezterm/config/lua/config/font_rules.html)), discussions and issues ([ex](https://github.com/wez/wezterm/discussions/1392)), Wezterm's source code, searching for dotfiles on Github ([ex](https://github.com/search?q=font_rules+intensity+half+language%3ALua+&type=code)), and tinkering with it, I can't get the behavior I want.

I would like for Wezterm to use a reduced opacity when encountering ANSI SGR 2 without affecting the font weight, font style, or color. So, for example, text that is 1;2;36 (bold, dim, and cyan), should be rendered using my cyan color and the bold font weight but with reduced opacity.

If this is possible with font_rules, could somebody please give me an example on how to achieve this? Thank you!

Big thanks to Wez and the other contributors for this amazing piece of software!! :blue_heart:

**Comment by wez** [comment, marked as answer]
If you use a font_rule to match on `intensity = 'Half'`, but explicitly use the same regular font weight, wezterm will synthesize dim text by rendering it with an hsv transform to set the brightness of the color to 50%

`wezterm ls-fonts` will tell you when that will happen; it will say something like "Will synthesize dim"

**Reply by justmoon** [reply]
Awesome, I will try that. Thanks!

**Reply by a-hariti** [reply]
Did it work?
I believe it is broken on latest
